central/testutils/testexport.go

```diff
@@ -0,0 +1,348 @@
+		}
+		nbContainers := (i % 3) + 1
+		containers := make([]*storage.Container, 0, nbContainers)
+		for j := 0; j < nbContainers; j++ {
+			container := &storage.Container{}
+			err := testutils.FullInit(container, testutils.UniqueInitializer(), testutils.JSONFieldsFilter)
+			if err != nil {
+				return err
+			}
+			idx := int(rand.Int31()) % len(imageIDs)
+			imageID := imageIDs[idx]
+			imageName := imageNamesByIDs[imageID]
+			containerImage := &storage.ContainerImage{
+				Id:             imageID,
+				Name:           imageName,
+				NotPullable:    false,
+				IsClusterLocal: false,
+			}
+			container.Image = containerImage
+			containers = append(containers, container)
+		}
+		deployment.Containers = containers
+		if i%10 == 9 {
+			deployment.Namespace = namespace10pct
+		} else {
+			deployment.Namespace = namepsace90pct
+		}
+		err = h.Deployments.UpsertDeployment(upsertCtx, deployment)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// InjectDataAndRunBenchmark pushes datasets of various sizes to database,
+// and runs the provided benchmark function against them.
+func (h *ExportServicePostgresTestHelper) InjectDataAndRunBenchmark(
+	b *testing.B,
+	injectImages bool,
+	benchmark func(b *testing.B),
+) {
+	datasetSizes := []int{500}
+	// The test runs by default with a lower scale as smoke test
+	// in the benchmark unit tests. To test at higher scales (takes time),
+	// run the test with ROX_SCALE_TEST set to a non-empty value
+	// in the test environment.
+	scale := os.Getenv("ROX_SCALE_TEST")
+	if scale != "" {
+		datasetSizes = []int{500, 1000, 2000, 5000, 10000}
+	}
+	imageIDs := make([]string, 0)
+	imageNamesByIDs := make(map[string]*storage.ImageName)
+	if !injectImages {
+		images, err := GetBaseImageSet()
+		if err != nil {
+			b.Error(err)
+		}
+		for _, image := range images {
+			imageID := image.GetId()
+			imageName := image.GetName()
+			imageIDs = append(imageIDs, imageID)
+			imageNamesByIDs[imageID] = imageName
+		}
+	}
+
+	slices.Sort(datasetSizes)
+	lastDatasetSize := 0
+	for ix, datasetSize := range datasetSizes {
+		delta := datasetSize - lastDatasetSize
+		if injectImages {
+			log.Info("Injecting ", delta, " images")
+			addedImageIDs, addedImageNamesByID, err := h.InjectImages(b, delta)
+			if err != nil {
+				b.Error(err)
+			}
+			imageIDs = append(imageIDs, addedImageIDs...)
+			for imageID, imageName := range addedImageNamesByID {
+				imageNamesByIDs[imageID] = imageName
+			}
+		}
+		log.Info("Injecting ", delta, " deployments")
+		err := h.InjectDeployments(b, delta, imageIDs, imageNamesByIDs)
+		if err != nil {
+			b.Error(err)
+		}
+		log.Info("Test iteration ", ix+1)
+		b.Run(fmt.Sprintf("%d", datasetSize), benchmark)
+	}
+
+}
+
+// TestScenario contains the parameters for an export API test.
+type TestScenario struct {
+	Name            string
+	TargetNamespace string
+}
+
+// GetBaseTestCases returns a basic set of TestScenario objects.
+func GetBaseTestCases() []TestScenario {
```

Comment:
Same here, since we're in a package that can theoretically hold more than just the export related test setup, it should be named more explicitly (i.e. `TestScenario` seems tied to the export tests, so it should be named accordingly; same applies to `BaseTestCases`).